### ISSUE TITLE
StableDropout class

### ISSUE THREAD

**Issue by fkrasnov2** [huggingface/transformers#33646]
### System Info

Hi,

Everything's going great with DeBERTa, thanks a lot! I've got one little thing though: 
The dropout value is a bit hard to read: it's StableDropout().  Could be much better:  StableDropout(p=0.1)

![image](https://github.com/user-attachments/assets/81384c9c-56ca-44a0-9614-d072d63fa328)

It's no big deal, but if you think there's something I can do about it, let me know.

Best,
Fedor (Ted)

### Who can help?

@ArthurZucker

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

https://github.com/huggingface/transformers/blob/78b2929c0554b79e0489b451ce4ece14d265ead2/src/transformers/models/deberta/modeling_deberta.py#L207

### Expected behavior

class StableDropout(nn.Module):
    """
    Optimized dropout module for stabilizing the training

    Args:
        drop_prob (float): the dropout probabilities
    """

    def __init__(self, drop_prob):
        super().__init__()
        self.drop_prob = drop_prob
        self.count = 0
        self.context_stack = None
    
    def __repr__(self):
        return f"StableDropout(p={self.drop_prob:.4f})"
        
    def __str__(self):
        return f"StableDropout(p={self.drop_prob:.4f})"

**Comment by LysandreJik** [comment]
Would you like to open a PR to fix this?

**Comment by fkrasnov2** [comment]
Yes, I do.
